src/envoy/http/service_control/http_call.h

```diff
@@ -43,20 +43,29 @@ class HttpCall {
 class HttpCallFactory
     : public Envoy::Logger::Loggable<Envoy::Logger::Id::filter> {
  public:
-  HttpCallFactory(Envoy::Upstream::ClusterManager& cm,
-                  Envoy::Event::Dispatcher& dispatcher,
-                  const ::espv2::api::envoy::v6::http::common::HttpUri& uri,
-                  const std::string& suffix_url,
-                  std::function<const std::string&()> token_fn,
-                  uint32_t timeout_ms, uint32_t retries,
-                  Envoy::TimeSource& time_source,
-                  const std::string& trace_operation_name);
+  virtual HttpCall* createHttpCall(const Envoy::Protobuf::Message& body,
+                                   Envoy::Tracing::Span& parent_span,
+                                   HttpCall::DoneFunc on_done) PURE;
+
+  virtual ~HttpCallFactory(){};
+};
+
+class HttpCallFactoryImpl : public HttpCallFactory {
```

Comment:
can we move this Impl into cc file?

Reply:
No, it should be in header. ClientCache creates a `HttpCallFactoryImpl`.